code_examples/dapp/src/dapp/dappIntroduction.ts

```diff
@@ -15,16 +15,19 @@ let window: {
 }
 
 export async function main() {
+  const api = Kilt.ConfigService.get('api')
+
   const did = 'did:kilt:4smcAoiTiCLaNrGhrAM4wZvt5cMKEGm8f3Cu9aFrpsh5EiNV'
   const dAppName = 'Your dApp Name'
 
-  const fullDid = await Kilt.Did.fetch(did)
+  const encodedFullDid = await api.call.did.query(Kilt.Did.toChain(did))
+  const { document } = Kilt.Did.linkedInfoFromChain(encodedFullDid)
   // If there is no DID, or the DID does not have any key agreement key, return
-  if (!fullDid || !fullDid.keyAgreement || !fullDid.keyAgreement[0]) {
+  if (!document || !document.keyAgreement || !document.keyAgreement[0]) {
```

Comment:
Since `document` is always defined, no need for this extra check here and for question mark below
```suggestion
  if (!document.keyAgreement || !document.keyAgreement[0]) {
```

Reply:
Good point: https://github.com/KILTprotocol/docs/pull/168/commits/f2ccc8f1b021d5164ce44f10cc3c7e93924cfd19